File: app/src/main/java/ru/nsu/fit/modao/RegistrationFragment.java
package ru.nsu.fit.modao;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import ru.nsu.fit.modao.databinding.FragmentRegistrationBinding;
import ru.nsu.fit.modao.model.NewUser;
import ru.nsu.fit.modao.repository.LoginViewModel;
import ru.nsu.fit.modao.repository.MyApplication;


public class RegistrationFragment extends Fragment {

    FragmentRegistrationBinding binding;
    LoginViewModel loginViewModel;
    BottomNavigationView bottomMenu;
    NavController navController;
    MyApplication app;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentRegistrationBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }
    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        navController = Navigation.findNavController(view);
        bottomMenu = getActivity().findViewById(R.id.bottomMenu);
        loginViewModel = new ViewModelProvider(this).get(LoginViewModel.class);
        app = (MyApplication) getActivity().getApplication();
        loginViewModel.getUserIn().observe(getViewLifecycleOwner(), new Observer<Long>() {
            @Override
            public void onChanged(Long id) {
                if (id == -1L){
                    binding.tipMessage.setText(R.string.alreadyExist);
                    return;
                }
                if (id == -2L){
                    binding.tipMessage.setText(R.string.serverProblems);
                    return;
                }
                app.setUserID(id);
                bottomMenu.setVisibility(View.VISIBLE);
                NavDirections action = RegistrationFragmentDirections.actionRegistrationFragmentToProfileFragment();
                navController.navigate(action);
            }
        });
        binding.buttonContinue.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String log = binding.createLogin.getText().toString();
                String pass = binding.createPassword.getText().toString();
                if (log.equals("") || pass.equals("")){
                    binding.tipMessage.setText(R.string.enterData);
                    return;
                }
                loginViewModel.createUser(app.getQuestApi(), new NewUser(binding.createUsername.getText().toString(),
                        log, pass, null, null, null));
            }
        });
    }
}

File: app/src/main/java/ru/nsu/fit/modao/fragments/RegistrationFragment.kt
package ru.nsu.fit.modao.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import com.google.android.material.bottomnavigation.BottomNavigationView
import dagger.hilt.android.AndroidEntryPoint
import ru.nsu.fit.modao.R
import ru.nsu.fit.modao.databinding.FragmentRegistrationBinding
import ru.nsu.fit.modao.utils.App
import ru.nsu.fit.modao.viewmodels.LoginViewModel
import javax.inject.Inject

@AndroidEntryPoint
class RegistrationFragment: Fragment() {
    private var _binding: FragmentRegistrationBinding? = null
    private val binding get() = _binding!!
    @Inject
    lateinit var app: App
    private var login: String? = null
    private var password: String? = null
    private val loginViewModel: LoginViewModel by viewModels()
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        _binding = FragmentRegistrationBinding.inflate(inflater, container, false)
        return binding.root
    }
    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        loginViewModel.message.observe(viewLifecycleOwner) {
            binding.tipMessage.text = it
        }
        loginViewModel.userId.observe(viewLifecycleOwner) {
            loginViewModel.login(login = login!!, password = password!!)
        }
        loginViewModel.token.observe(viewLifecycleOwner) {
            app.userId = it.id
            app.accessToken = it.accessToken
            app.refreshToken = it.refreshToken
            activity?.findViewById<BottomNavigationView>(R.id.bottomMenu)?.visibility = View.VISIBLE
            findNavController().navigate(RegistrationFragmentDirections.actionRegistrationFragmentToProfileFragment())
        }
        binding.buttonContinue.setOnClickListener {
            binding.tipMessage.setText(R.string.wait)
            password = binding.createPassword.text.toString()
            login = binding.createLogin.text.toString()
            loginViewModel.createUser(login!!, password!!,
            binding.createUsername.text.toString())
        }
    }
}